<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Run workflow to decompose tumor subclones"><title>Run workflow to decompose tumor subclones — run_numbat • numbat</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run workflow to decompose tumor subclones — run_numbat"><meta property="og:description" content="Run workflow to decompose tumor subclones"><meta property="og:image" content="https://kharchenkolab.github.io/numbat/logo.png"><meta name="google-site-verification" content="GE78cdBqx5YPEbHjqtohe_p51Ng8oEPXdcaSEIu2dP8"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">numbat</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/numbat.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/results.html">Interpreting Numbat results</a>
    <a class="dropdown-item" href="../articles/mouse.html">Mouse tutorial</a>
    <a class="dropdown-item" href="../articles/spatial-rna.html">Spatial transcriptomics</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Run workflow to decompose tumor subclones</h1>
      
      <div class="d-none name"><code>run_numbat.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run workflow to decompose tumor subclones</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_numbat</span><span class="op">(</span></span>
<span>  <span class="va">count_mat</span>,</span>
<span>  <span class="va">lambdas_ref</span>,</span>
<span>  <span class="va">df_allele</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  max_iter <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  max_nni <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  t <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  gamma <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  min_LLR <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">1e-04</span>,</span>
<span>  eps <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  max_entropy <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  init_k <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  min_cells <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  tau <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  nu <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  max_cost <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span> <span class="op">*</span> <span class="va">tau</span>,</span>
<span>  n_cut <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  min_depth <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  common_diploid <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  min_overlap <span class="op">=</span> <span class="fl">0.45</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  ncores_nni <span class="op">=</span> <span class="va">ncores</span>,</span>
<span>  random_init <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  segs_loh <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  diploid_chroms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  use_loh <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_genes <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  skip_nj <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  multi_allelic <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  p_multi <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">alpha</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  check_convergence <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  exclude_neu <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>count_mat</dt>
<dd><p>dgCMatrix Raw count matrices where rownames are genes and column names are cells</p></dd>


<dt>lambdas_ref</dt>
<dd><p>matrix Either a named vector with gene names as names and normalized expression as values, or a matrix where rownames are genes and columns are pseudobulk names</p></dd>


<dt>df_allele</dt>
<dd><p>dataframe Allele counts per cell, produced by preprocess_allele</p></dd>


<dt>genome</dt>
<dd><p>character Genome version (hg38, hg19, or mm10)</p></dd>


<dt>out_dir</dt>
<dd><p>string Output directory</p></dd>


<dt>max_iter</dt>
<dd><p>integer Maximum number of iterations to run the phyologeny optimization</p></dd>


<dt>max_nni</dt>
<dd><p>integer Maximum number of iterations to run NNI in the ML phylogeny inference</p></dd>


<dt>t</dt>
<dd><p>numeric Transition probability</p></dd>


<dt>gamma</dt>
<dd><p>numeric Dispersion parameter for the Beta-Binomial allele model</p></dd>


<dt>min_LLR</dt>
<dd><p>numeric Minimum LLR to filter CNVs</p></dd>


<dt>alpha</dt>
<dd><p>numeric P value cutoff for diploid finding</p></dd>


<dt>eps</dt>
<dd><p>numeric Convergence threshold for ML tree search</p></dd>


<dt>max_entropy</dt>
<dd><p>numeric Entropy threshold to filter CNVs</p></dd>


<dt>init_k</dt>
<dd><p>integer Number of clusters in the initial clustering</p></dd>


<dt>min_cells</dt>
<dd><p>integer Minimum number of cells to run HMM on</p></dd>


<dt>tau</dt>
<dd><p>numeric Factor to determine max_cost as a function of the number of cells (0-1)</p></dd>


<dt>nu</dt>
<dd><p>numeric Phase switch rate</p></dd>


<dt>max_cost</dt>
<dd><p>numeric Likelihood threshold to collapse internal branches</p></dd>


<dt>n_cut</dt>
<dd><p>integer Number of cuts on the phylogeny to define subclones</p></dd>


<dt>min_depth</dt>
<dd><p>integer Minimum allele depth</p></dd>


<dt>common_diploid</dt>
<dd><p>logical Whether to find common diploid regions in a group of peusdobulks</p></dd>


<dt>min_overlap</dt>
<dd><p>numeric Minimum CNV overlap threshold</p></dd>


<dt>ncores</dt>
<dd><p>integer Number of threads to use</p></dd>


<dt>ncores_nni</dt>
<dd><p>integer Number of threads to use for NNI</p></dd>


<dt>random_init</dt>
<dd><p>logical Whether to initiate phylogney using a random tree (internal use only)</p></dd>


<dt>segs_loh</dt>
<dd><p>dataframe Segments of clonal LOH to be excluded</p></dd>


<dt>verbose</dt>
<dd><p>logical Verbosity</p></dd>


<dt>diploid_chroms</dt>
<dd><p>vector Known diploid chromosomes</p></dd>


<dt>use_loh</dt>
<dd><p>logical Whether to include LOH regions in the expression baseline</p></dd>


<dt>min_genes</dt>
<dd><p>integer Minimum number of genes to call a segment</p></dd>


<dt>skip_nj</dt>
<dd><p>logical Whether to skip NJ tree construction and only use UPGMA</p></dd>


<dt>multi_allelic</dt>
<dd><p>logical Whether to call multi-allelic CNVs</p></dd>


<dt>p_multi</dt>
<dd><p>numeric P value cutoff for calling multi-allelic CNVs</p></dd>


<dt>plot</dt>
<dd><p>logical Whether to plot results</p></dd>


<dt>check_convergence</dt>
<dd><p>logical Whether to terminate iterations based on consensus CNV convergence</p></dd>


<dt>exclude_neu</dt>
<dd><p>logical Whether to exclude neutral segments from CNV retesting (internal use only)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>a status code</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Teng Gao, Ruslan Soldatov, Hirak Sarkar, Evan Biederstedt, Peter Kharchenko.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

